### PR TITLE
implement --clean flag for build command

### PR DESCRIPTION
**Pull request by appeltel**
Implemented "--clean / -c" flag for build command which removes an existing venv. Currently this just performs an "rm -r" on the venv_path if it exists and nothing otherwise.

### CODE REVIEW

**Comment by evansde77** [comment]
LGTM

File: src/cirrus/build.py [hunk 1]
```diff
@@ -45,6 +67,13 @@ def main():
         'venv',
         'bin',
         'virtualenv')
+
+    # remove existing virtual env if building clean
+    if opts.clean is True and os.path.exists(venv_path):
+        cmd = "rm -r {0}".format(venv_path)
```
Comment (jcounts) [inline src/cirrus/build.py]: maybe `-rf`?

**Comment by jcounts** [comment]
Small comment but +1 overall. Convenient little addition :-)

File: src/cirrus/build.py [hunk 2]
```diff
@@ -11,21 +11,43 @@
...
-    _main_
+    _build_parser_
+
+    Set up command line parser for the build command
+
+    """
+    parser = ArgumentParser(
+        description = 'git cirrus build'
+    )
+    parser.add_argument('command',nargs='?')
```
Comment (ksnavely) [inline src/cirrus/build.py]: Make sure you are either running pep8 or have a pep8 plugin running in your editor (space after comma)

File: src/cirrus/build.py [hunk 3]
```diff
@@ -11,21 +11,43 @@
...
+def build_parser(argslist):
     """
-    _main_
+    _build_parser_
+
+    Set up command line parser for the build command
+
+    """
+    parser = ArgumentParser(
+        description = 'git cirrus build'
```
Comment (ksnavely) [inline src/cirrus/build.py]: pep8

File: src/cirrus/build.py [hunk 4]
```diff
@@ -11,21 +11,43 @@
...
+    _execute_build_
 
     Execute the build in the current package context.
 
     - reads the config to check for custom build parameters
       - defaults to ./venv for virtualenv
       - defaults to ./requirements.txt for reqs
+    - removes existing virtualenv if clean flag is set
     - builds the virtualenv
     - pip installs the requirements into it
```
Comment (ksnavely) [inline src/cirrus/build.py]: Could you document `opts` and its type? We typically use sphinx style `:param FooType opts: A FooType of blah`

File: src/cirrus/build.py (hunk 1 above)
Comment (ksnavely) [inline src/cirrus/build.py]: The `f` is only needed if you do not have permissions to change the file, which you should have in your local dev environment and I believe we'll have in the CI environment.

File: src/cirrus/build.py [hunk 5]
```diff
@@ -45,6 +67,13 @@ def main():
         'venv',
         'bin',
         'virtualenv')
+
+    # remove existing virtual env if building clean
+    if opts.clean is True and os.path.exists(venv_path):
```
Comment (ksnavely) [inline src/cirrus/build.py]: `if opts.clean` is probably more idiomatic than `if opts.clean is True`

File: src/cirrus/build.py [hunk 6]
```diff
@@ -45,6 +67,13 @@ def main():
...
+        cmd = "rm -r {0}".format(venv_path)
+        print "Removing existing virtualenv: {0}".format(venv_path)
```
Comment (ksnavely) [inline src/cirrus/build.py]: You should use the logger. https://github.com/evansde77/cirrus/blob/b13e908690c1c58b244ce1269f1bfccc62f052ae/src/cirrus/deploy.py#L18

In general one should avoid print (if logging facilities are provided); if you do need to use it then use the functional form `print(...)`.

File: src/cirrus/build.py [hunk 7]
```diff
@@ -11,21 +11,43 @@
...
 import sys
+from argparse import ArgumentParser
 
 from cirrus.environment import cirrus_home
 from cirrus.configuration import load_configuration, get_pypi_auth
 from fabric.operations import local
 
 
-def main():
+def build_parser(argslist):
```
Comment (ksnavely) [inline src/cirrus/build.py]: Document argslist in the docstring?

**Comment by ksnavely** [comment]
+1 with some minor nits

File: src/cirrus/build.py (hunk 1 above)
Comment (ksnavely) [inline src/cirrus/build.py]: Woops, that's wrong. Lesson: Do not review code at 10:30 at night. You may indeed want the f option.

Comment (shudgston) [inline src/cirrus/build.py]: You may want to try out https://docs.python.org/2/library/shutil.html#shutil.rmtree, just to remove os-specific commands.

Comment (shudgston) [inline src/cirrus/build.py]: Maybe you are fine.  Didn't see the fabric stuff until now.  I'll let others comment on this.

File: src/cirrus/build.py (hunk 5 above)
Comment (shudgston) [inline src/cirrus/build.py]: It is

File: src/cirrus/build.py (hunk 1 above)
Comment (ksnavely) [inline src/cirrus/build.py]: A good thing to keep in mind in general; the scope of cirrus is probably for *nix flavored OS's. Pragmatically this is OS X and Linux.

**Comment by appeltel** [comment]
Ok, thanks for the corrections. I added a force flag to rm, but kept with fabric for consistency, added shpinx-style parameters to the docstrings, actually ran pep8 (oops!), removed the "is True", and made the module use the logger.

**Pull request merged**